include/fastdds/rtps/common/CDRMessage_t.h

```diff
@@ -165,6 +165,24 @@ struct RTPS_DllAPI CDRMessage_t{
         return *(this);
     }
 
+    void init(
+            octet* buffer_ptr,
+            uint32_t size)
+    {
+        assert(buffer == nullptr);
+        wraps = true;
+        pos = 0;
+        length = 0;
+        buffer = buffer_ptr;
+        max_size = size;
```

Comment:
After #974 is merged we should rebase this PR and add one line here with `reserved_size = size;`